plugins/filestorage/localfs/index.js

```diff
@@ -290,6 +291,57 @@ LocalFileStorage.prototype.getFileStats = function (filePath, callback) {
 };
 
 /**
+ * Copies an asset from one tenant to another.
+ * 
+ * @param {object} asset - A valid 'asset' record
+ * @param {string} sourceTenantName - Source tenant.name
+ * @param {string} destinationTenantName - Destination tenant.name
+ */
+LocalFileStorage.prototype.copyAsset = function(asset, sourceTenantName, destinationTenantName, callback) {
+  // Verify the destination folder exists
+  var assetFolder = path.join(configuration.serverRoot, configuration.getConfig('dataRoot'), destinationTenantName, asset.directory);
+
+  mkdirp(assetFolder, function(error) {
+    if (error) {
+      logger.log('error', 'mkdirp error while copying asset, creating ' + assetFolder);
+      return callback(error);
+    }
+
+    // Copy the asset
+    var sourceFile = path.join(configuration.serverRoot, configuration.getConfig('dataRoot'), sourceTenantName, asset.path);
+    var destinationFile = path.join(configuration.serverRoot, configuration.getConfig('dataRoot'), destinationTenantName, asset.path);
+
+    ncp(sourceFile, destinationFile, {clobber:false}, function(error) {
+      if (error) {
+        logger.log('error', 'ncp error while copying ' + asset.filename);
+        logger.log('error', 'error copying ' + sourceFile + ' to ' + destinationFile);
+        return callback(error);
+      }
+
+      // Copy the thumbnail (if required)
+      if (asset.thumbnailPath && asset.thumbnailPath !== 'false' && (asset.assetType == 'image' || asset.assetType == 'video')) {
+
+        var sourceThumbnailFile = path.join(configuration.serverRoot, configuration.getConfig('dataRoot'), sourceTenantName, asset.thumbnailPath);
+        var destinationThumbnailFile = path.join(configuration.serverRoot, configuration.getConfig('dataRoot'), destinationTenantName, asset.thumbnailPath);
+
+        ncp(sourceThumbnailFile, destinationThumbnailFile, function(error) {
+          if (error) {
+            logger.log('error', 'ncp error while copying ' + asset.thumbnailPath);
+            logger.log('error', 'error copying (thumbnail) ' + sourceThumbnailFile + ' to ' + destinationThumbnailFile);
+
+            // TODO If thumbnails fail it's not the end of the world -- for now
```

Comment:
would it be reasonable to call [`createThumbnail()`](https://github.com/adaptlearning/adapt_authoring/pull/1058/files#diff-75d6b374d5cc336322b6494e003f6e1eR353) here if the copy of the thumb fails?